pkg/backup/restore_job.go

```diff
@@ -724,8 +724,21 @@ func remapAndFilterRelevantStatistics(
 		if statShouldBeIncludedInBackupRestore(stat) {
 			tableHasStatsInBackup[stat.TableID] = struct{}{}
 			if tableRewrite, ok := descriptorRewrites[stat.TableID]; ok {
-				// Statistics imported only when table re-write is present.
+				// We only restore statistics when all necessary descriptor rewrites are
+				// present in the rewrite map.
 				stat.TableID = tableRewrite.ID
+				// We also need to remap the type OID in the histogram for UDTs.
+				if stat.HistogramData != nil && stat.HistogramData.ColumnType != nil {
+					if typ := stat.HistogramData.ColumnType; typ.UserDefined() {
+						typDescID := typedesc.GetUserDefinedTypeDescID(typ)
+						if _, ok := descriptorRewrites[typDescID]; !ok {
+							continue
```

Comment:
We can still restore statistics if the UDT is not in the rewrite map?

Reply:
Maybe I'm misunderstanding your question... the `continue` should prevent the stats from being restored if the UDT is not in the rewrite map. And I think this is what we want, since otherwise we won't know what the UDT OID should be.

Reply:
Ahh, I misread the code. It might be time to convert the 5-level nested ifs into some early `continue` ifs instead... 😆